src/views/studio/EditVideoSheet/EditVideoSheet.tsx

```diff
@@ -58,25 +73,31 @@ export const EditVideoSheet: React.FC = () => {
       isExplicit: null,
     },
   })
+  const onSubmit = handleSubmit((data) => {
+    console.log(data)
+  })
+
+  const resetFields = (video: VideoDraft | null) => ({
+    title: video?.title,
+    description: video?.description,
+    selectedVideoVisibility: video?.isPublic === undefined ? null : video.isPublic ? 'public' : 'unlisted',
+    selectedVideoLanguage: video?.language ?? null,
+    selectedVideoCategory: video?.categoryId ?? null,
+    hasMarketing: video?.hasMarketing ?? null,
+    publishedBeforeJoystream: video?.publishedBeforeJoystream ?? '',
+    isExplicit: video?.isExplicit === undefined ? null : video.isExplicit,
+  })
 
   // Tabs
-  const { addDraft } = useDrafts('video', channelId)
+  const { addDraft, drafts, getDraft, updateDraft } = useDrafts('video', channelId)
 
   const selectTab = useCallback(
-    (tab: EditVideoSheetTab) => {
+    async (tab: EditVideoSheetTab) => {
+      const currentDraft = await getDraft(tab.id)
```

Comment:
I don't think we need to fetch the draft directly from the local storage here. The hook exposes `drafts` which should be a list of drafts , we can use that.